src/CorePolicy.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef COREPOLICY_H_
+#define COREPOLICY_H_
+
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#include <functional>
+#include <vector>
+#include <mutex>
+#include <deque>
+#include <atomic>
+#include <queue>
+#include <string>
+#include <thread>
+
+#include "PerfUtils/Cycles.h"
+#include "PerfUtils/Util.h"
+#include "Logger.h"
+#include "PerfStats.h"
+#include "Common.h"
+
+/*
+ * Resizing of the threadClassCoreMap is not thread-safe and cannot be made
+ * thread-safe in a performant way, so for simplicity the threadClassCoreMap
+ * is allocated to a fixed size with MAX_THREAD_CLASSES threadClass slots, 
+ * which should be more than are ever needed.
+ */
+#define MAX_THREAD_CLASSES 64
+
+typedef uint32_t threadClass_t;

```

Comment:
General question: does this code review also address my comments that weren't related to documentation? I see several things that I thought I commented on, but which don't appear to have changed, like the name of this class. Let's discuss.